provider_test.go

```diff
@@ -293,6 +293,7 @@ func TestSetRecords(t *testing.T) {
 	ctx := setup(t, `[
 		{"subname": "www", "type": "A", "ttl": 3600, "records": ["127.0.1.1", "127.0.1.2"]},
 		{"subname": "", "type": "TXT", "ttl": 3600, "records": ["\"will be overridden\""]},
+		{"subname": "sub", "type": "TXT", "ttl": 3600, "records": ["\"will stay the same\""]},
```

Comment:
We should also test that records with the same name but a different type to the ones being set are not deleted. We haven't used NS as type yet, so adding it here should do the trick.

One issue I see is that the test is getting a bit lengthy and it's testing a number of cases at once that should better be separated, but that's something I can do as a follow-up (i.e. rewrite as a table driven test).

Reply:
Added an NS record to the test.